### PR TITLE
api: Fix pagination for list PublicIPAddresses

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description

Fixes: https://github.com/apache/cloudstack/issues/5229
This PR fixes the issue noticed while listing public IP addresses wrt pagination. The count is set to the number of items being returned as opposed to the total number of entities (here, public IPs)

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?
via cmk:

Prior Fix:
```
## Total number of ip addresses returned without pagination
list publicipaddresses  listall=true filter=id
{
  "count": 55,
  "publicipaddress": [
    {
       ....
    }
    ]
}    

## With Pagination

 list publicipaddresses  pagesize=5 page=1 listall=true filter=id
{
  "count": 5,    <--- while count has to be 55, it returns 5 which is the value of pagesize
  "publicipaddress": [
    {
      "id": "152f39af-1af7-4413-bbf1-1132b1708a9d"
    },
    {
      "id": "cbd3f921-53b5-45a6-9a1b-8c5877e9be34"
    },
    {
      "id": "425204aa-8efd-400a-bc07-98a14fae00b3"
    },
    {
      "id": "dc5b206d-db46-4ab5-9ede-da63261648e5"
    },
    {
      "id": "437fb07f-2c9e-4583-805b-710474c045be"
    }
  ]
}

```
Post Fix:
```
## Total number of addresses without pagination:
SBCM5> > list publicipaddresses listall=true filter=id
{
  "count": 20,
  "publicipaddress": [
    {
      ....
    }
  ]
}
  
## With Pagination:  
SBCM5> > list publicipaddresses listall=true filter=id, page=1 pagesize=5
{
  "count": 20,  <--- Returns the actual number of public IP addresses
  "publicipaddress": [
    {
      "id": "a000effc-0dea-4a84-8299-2de2e38b6d79"
    },
    {
      "id": "c0d594a4-bca2-43c2-bbb2-32ab9eea13a4"
    },
    {
      "id": "d270125d-9578-4d56-9453-96a80c00e721"
    },
    {
      "id": "8fe27d41-ab1a-41e2-8e24-0c78ab389603"
    },
    {
      "id": "04466b5d-b61f-4ff2-9d41-c4354966acf4"
    }
  ]
}

```

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/server/ManagementServerImpl.java [hunk 1]
```diff
@@ -2197,7 +2197,11 @@ private void abstractDataStoreClustersList(List<StoragePool> storagePools, List<
             sc2.setParameters("ids", freeAddrIds.toArray());
             addrs.addAll(_publicIpAddressDao.search(sc2, searchFilter)); // Allocated + Free
         }
-
+        List<? extends IpAddress> wPagination = com.cloud.utils.StringUtils.applyPagination(addrs, cmd.getStartIndex(), cmd.getPageSizeVal());
+        if (wPagination != null) {
+            Pair<List<? extends IpAddress>, Integer> listWPagination = new Pair<List<? extends IpAddress>, Integer>(wPagination, addrs.size());
+            return listWPagination;
```
Comment (shwstppr) [inline server/src/main/java/com/cloud/server/ManagementServerImpl.java:2203]:
```suggestion
            return new Pair<>(wPagination, addrs.size());
```
shorter

**Review by shwstppr — APPROVED** [review]
LGTM. Tested for different cases, correct items and count returned by API.

```
(localcloud) 🐦 > list publicipaddresses listall=true page=1 pagesize=2
{
  "count": 3,
...
```
```
(localcloud) 🐧 > list publicipaddresses listall=true page=1 pagesize=2 ipaddress=192.168.2.3 
{
  "count": 1,
  "publicipaddress": [
...
```
```
(localcloud) 🦍 > list publicipaddresses listall=true page=1 pagesize=2 allocatedonly=true 
{
  "count": 3,
...
```
```
(localcloud) 🦍 > list publicipaddresses listall=true page=1 pagesize=15 allocatedonly=false filter=id
{
  "count": 395,
...
```

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test

**Review by nvazquez — APPROVED** [review]
Code LGTM

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Pull request merged**